tests/vmi_multus_test.go

```diff
@@ -938,6 +952,34 @@ var _ = Describe("[Serial]Macvtap", func() {
 		return vmi
 	}
 
+	createRunningVMIWithMacOnPrimaryAddr := func(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance, mac string, expecterFactory console.VMIExpecterFactory) *v1.VirtualMachineInstance {
+		vmi.Spec.Domain.Devices.Interfaces[primaryIfaceIndex].MacAddress = mac
```

Comment:
Could you extend libvmi with MacAddress attribute instead? This would fit into the builder pattern used couple of lines above.

Reply:
I could, but why would I ? The moment it becomes a necessity elsewhere, let that someone (which I'm pretty sure will be future me do that).

Furthermore, to which address would it set the mac ?... Is primary / secondary good enough ? What if we have multiple secondary ifaces ? I don't care about any of those scenarios.

I seriously don't want this PR to hang indefinitely until we decide what is the API of an helper method I have nothing to gain from moving into a library.

Reply:
> The moment it becomes a necessity elsewhere, let that someone (which I'm pretty sure will be future me do that).

What do you mean by this?

You would get easier access to my reviews FWIW since I would be happier parsing code from a builder library I know rather than bunch of ad-hoc functions.

The library was supposed to make building of these helpers easier. If it fails to deliver that, we have a problem somewhere.

Isn't all you need `libvmi.WithInterfaceWithMAC`? Or better, a builder for the interface object passd to `libvmi.WithInterface`.

Pardon me being so blunt, but arguing that you want to get it merged fast or that you don't care about the abstraction is not convincing me to merge this as is, quite the opposite.

Reply:
> > The moment it becomes a necessity elsewhere, let that someone (which I'm pretty sure will be future me do that).
> 
> What do you mean by this?

I meant that until something proves to be a pattern there's no point in making it a reusable component. 

While I'm aware that an interface with a mac address is something that happens a lot within that test, refactoring this test is not in my list of goals / tasks. Creating a functional test that shows that you can migrate a VMI from one node to another is - and I'd like to keep the scope of this PR to that. 

> 
> You would get easier access to my reviews FWIW since I would be happier parsing code from a builder library I know rather than bunch of ad-hoc functions.

ACK.

> 
> The library was supposed to make building of these helpers easier. If it fails to deliver that, we have a problem somewhere.
> 
> Isn't all you need `libvmi.WithInterfaceWithMAC`? Or better, a builder for the interface object passd to `libvmi.WithInterface`.

Good suggestion.

> 
> Pardon me being so blunt, but arguing that you want to get it merged fast or that you don't care about the abstraction is not convincing me to merge this as is, quite the opposite.

I never said any of that; I said, and I quote: "I seriously don't want this PR to hang indefinitely until we decide what is the signature of an helper function I have nothing to gain from moving into a library." . 

And I said that because the other macvtap PR I worked on hanged on for nearly a year (on unrelated matter, I know). Can't blame me for trying to have the scope - **and** changes - as small as possible on something that is not inside any of our sprints.

Reply:
> > > The moment it becomes a necessity elsewhere, let that someone (which I'm pretty sure will be future me do that).
> > 
> > 
> > What do you mean by this?
> 
> I meant that until something proves to be a pattern there's no point in making it a reusable component.
> 
> While I'm aware that an interface with a mac address is something that happens a lot within that test, refactoring this test is not in my list of goals / tasks. Creating a functional test that shows that you can migrate a VMI from one node to another is - and I'd like to keep the scope of this PR to that.
> 
> > You would get easier access to my reviews FWIW since I would be happier parsing code from a builder library I know rather than bunch of ad-hoc functions.
> 
> ACK.
> 
> > The library was supposed to make building of these helpers easier. If it fails to deliver that, we have a problem somewhere.
> > Isn't all you need `libvmi.WithInterfaceWithMAC`? Or better, a builder for the interface object passd to `libvmi.WithInterface`.
> 
> Good suggestion.

Let me know if you like https://github.com/kubevirt/kubevirt/pull/4412/commits/f0982a4a8b90c2f4def4b33c75a99a0a0c845eac . If so, I'll squash it into the previous one.

> 
> > Pardon me being so blunt, but arguing that you want to get it merged fast or that you don't care about the abstraction is not convincing me to merge this as is, quite the opposite.
> 
> I never said any of that; I said, and I quote: "I seriously don't want this PR to hang indefinitely until we decide what is the signature of an helper function I have nothing to gain from moving into a library." .
> 
> And I said that because the other macvtap PR I worked on hanged on for nearly a year (on unrelated matter, I know). Can't blame me for trying to have the scope - **and** changes - as small as possible on something that is not inside any of our sprints.

Reply:
I like it. Please go ahead with the squash.

Reply:
Done.

Reply:
How comes we still need to set it here?

Reply:
Hope it is better now.